Scripts/Python/clftImager.py

```diff
@@ -764,7 +772,11 @@ def OnTimer(self,id):
             imagerBtn.enableActivator()
             ImagerBtnVisible.run(self.key)
             PtEnableForwardMovement()
-            PtSendKIMessage(kEnableEntireYeeshaBook,0)
+            vault = ptVault()
+            start = vault.findChronicleEntry("StartPathChosen")
+            entryCleft = vault.findChronicleEntry("CleftSolved")
+            if not (start.chronicleGetValue() == "cleft" and entryCleft is None):
+                PtSendKIMessage(kEnableEntireYeeshaBook, 0)
```

Comment:
This anti-pattern has been repeated at least three times now. I'd suggest splitting it out into a function to reduce code duplication.